### ISSUE TITLE
SinglefileData should have a read() function

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida_core#2430]
It's even mentioned in the comments of the file
https://github.com/aiidateam/aiida_core/blob/3e96d166deeb15df1a6ed12850ac504ec310ea92/aiida/orm/data/singlefile.py#L16
but it seems there is no such method.

Reading the content of a file seems like the most basic operation one could imagine, so I think we need to offer it - happy to discuss the name, how, etc.

**Comment by sphuber** [comment]
I just came across this PR and realized that this has been addressed in PR #2504 that prepared the node interface for the future repository API where it is no longer necessarily a file system based solution. As a result, the `SingleFileData` class now has the `open` and `get_content` method, which return a file handle or the actual content for the file, respectively. If you are happy with that solution, we can probably close this issue.
